src/MongoDB.Driver.Core/Core/Connections/BinaryConnection.cs

```diff
@@ -277,6 +281,13 @@ private void OpenHelper(CancellationToken cancellationToken)
                 _stream = _streamFactory.CreateStream(_endPoint, cancellationToken);
                 helper.InitializingConnection();
                 _description = _connectionInitializer.InitializeConnection(this, cancellationToken);
+
+                // When compressing, clients MUST use the first compressor in the client's configured compressors list
+                // that is also in the servers list.
+                _sendCompressorType = _description.AvailableCompressors.Any()
+                    ? _description.AvailableCompressors.First()
+                    : (CompressorType?)null;
```

Comment:
Use a helper method:

                _sendCompressorType = ChooseSendCompressorTypeIfAny(_description);

Where the helper method is:

        private CompressorType? ChooseSendCompressorTypeIfAny(ConnectionDescription connectionDescription)
        {
            var availableCompressors = connectionDescription.AvailableCompressors;
            return availableCompressors.Count > 1 ? (CompressorType?)availableCompressors[0] : null;
        }